### ISSUE TITLE
[Bug]: Invalid virtaulenv created by`py_venv` 

### ISSUE THREAD

**Issue by liningpan** [aspect-build/rules_py#277]
### What happened?

Based on my testing, in `0.5.0` python 3.11 works in general, but the virtual environment created by `py_venv` still had the same issues as reported in #159, https://github.com/aspect-build/rules_py/issues/159#issuecomment-1931446119.

In `0.6.0` (thanks Alex for creating a release for me to test), the python interpreter is able to start, but neither packages from pip nor in the bazel workspace can be loaded. The `first_party.pth` file looks like this 

```text
../../../..
../../../../site-packages
../../../../site-packages
../../../../site-packages
../../../../site-packages
../../../../site-packages
../../../../site-packages
../../../../site-packages
../../../../
../../../../
```

The same behavior is also reported here https://github.com/aspect-build/rules_py/pull/226#issuecomment-1833958996.

I don't know if this will be fixed by the rust venv work. I decided to create this issues since the recurring issue is now spread across multiple threads. It might also be helpful to have this feature tested in CI.

### Version

Development (host) and target OS/architectures: Arch Linux x86

Output of `bazel --version`: 6.5.0

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file: 

- rules_python: 0.31.0
- aspect_rules_py: 0.5.0 / 0.6.0

Language(s) and/or frameworks involved:

- Python 3.11

**Comment by mattem** [comment]
0.6.0 is the commit before the refactor. I can't repo at HEAD

```
$ bazel run //:venv
INFO: Analyzed target //:venv (3 packages loaded, 2207 targets configured).
...
INFO: Running command line: bazel-bin/venv

$ /Users/matt/workspace/rules_py/.venv/bin/python
Python 3.11.7 (main, Jan  7 2024, 23:23:46) [Clang 16.0.3 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import site
>>> site.getsitepackages()
['/Users/matt/workspace/rules_py/.venv/lib/python3.11/site-packages']

$ .venv/bin/python
Python 3.11.7 (main, Jan  7 2024, 23:23:46) [Clang 16.0.3 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import site
>>> site.getsitepackages()
['/Users/matt/workspace/rules_py/.venv/lib/python3.11/site-packages']
```

Generated `.pth` file contains full paths when generating outside the sandbox for action,

```
$ cat .venv/lib/python3.11/site-packages/venv.venv.pth
/private/var/tmp/_bazel_matt/442be3b1483cc77971c0d7bfbd105b2d/execroot/aspect_rules_py/bazel-out/darwin_arm64-fastbuild/bin/venv.runfiles/.
/private/var/tmp/_bazel_matt/442be3b1483cc77971c0d7bfbd105b2d/execroot/aspect_rules_py/bazel-out/darwin_arm64-fastbuild/bin/venv.runfiles/aspect_rules_py
```

**Comment by liningpan** [comment]
Oh that looks much better. Do you guys have an ETA for the refactor?

**Comment by mattem** [comment]
It's landed at HEAD, so you could dep on that instead. 
We are currently working through releasing the binaries needed for the toolchains, they are currently just committed to the repo here, so if you don't mind that can give it a try now, no rust toolchain should be needed.

**Comment by liningpan** [comment]
I just tested v0.7.1 and it appears to be working. It would be nice to document glibc requirement somewhere, as it won't run on centos7 (which is reasonable at this point).
